### PR TITLE
Enable Checkstyle and FindBugs in TonY #166

### PR DESCRIPTION
**Pull request by erwa**
* Enabled Checkstyle and FindBugs plugins.
* Configured build to fail if there are ANY Checkstyle or FindBugs warnings or errors.
* Fixed all Checkstyle and FindBugs warnings and errors.

### CODE REVIEW

**Comment by erwa** [comment]
Fixed test failures.

File: tony-core/src/main/java/com/linkedin/tony/TaskExecutor.java [hunk 1]
```diff
@@ -303,30 +304,7 @@ public HdfsAvroFileSplitReader getHdfsAvroFileSplitReader(List<String> readPaths
     return new HdfsAvroFileSplitReader(hdfsConf, readPaths, this.taskIndex,
         this.numTasks, useRandomShuffle);
   }
-
-
-  //endregion
-
-  //region Testing
-
-  private void hangIfTesting() {
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/TaskExecutor.java]: is this no longer needed?

Comment (erwa) [reply tony-core/src/main/java/com/linkedin/tony/TaskExecutor.java]: It's no longer needed. I deleted the test that used it in https://github.com/linkedin/TonY/pull/69 but forgot to delete this method.

File: tony-cli/src/main/java/com/linkedin/tony/cli/NotebookSubmitter.java [hunk 2]
```diff
@@ -81,9 +78,6 @@ public int submit(String[] args) throws ParseException, URISyntaxException, IOEx
 
     client.init(updatedArgs);
     client.start();
-    if (client == null) {
```
[the remark this answers is not in the document]
Comment (erwa) [reply tony-cli/src/main/java/com/linkedin/tony/cli/NotebookSubmitter.java]: The constructor initializes `client`:
```
  private NotebookSubmitter() {
    this.client = new TonyClient(new Configuration());
  }
```

Plus, if `client` were `null`, the line right above the line I deleted would have already thrown a `NullPointerException`.

**Pull request merged**